src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php

```diff
@@ -670,6 +715,18 @@ private function updateData(array $data, string $attribute, mixed $attributeValu
             return $data;
         }
 
+        if ($this->classMetadataFactory && $this->classMetadataFactory->hasMetadataFor($class)) {
+            $classMetadata = $this->classMetadataFactory->getMetadataFor($class);
+            $attributesMetadata = $classMetadata->getAttributesMetadata();
```

Comment:
better pass $attributesMetadata as argument instead of getting it again and again (it's already found in the caller)